### PR TITLE
Moar actions iii

### PR DESCRIPTION
**Pull request by Carreau**
on top of #689 add the following actions:

            - 'toggle-all-cells-output-visibility',
            - 'toggle-all-cells-output-scrolling',
            - 'clear-all-cells-output',

Ping @ellisonbg for the naming

### CODE REVIEW

**Comment by Carreau** [comment]
rebased to force github to refresh the diff.

**Comment by Carreau** [comment]
Ping @jhamrick who might help for naming too.

**Comment by jhamrick** [comment]
Maybe `cell` instead of `cells`, but I think either is fine, really.

**Comment by Carreau** [comment]
> Maybe cell instead of cells, but I think either is fine, really.

for the 3 of them ?

Re-ping @ellisonbg

**Comment by ellisonbg** [comment]
I should be able to look at this tomorrow

On Tue, Nov 3, 2015 at 2:16 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> Maybe cell instead of cells, but I think either is fine, really.
>
> for the 3 of them ?
>
> Re-ping @ellisonbg <https://github.com/ellisonbg>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/692#issuecomment-153458944>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

File: notebook/static/notebook/js/actions.js [hunk 1]
```diff
@@ -563,6 +563,25 @@ define(function(require){
...
+            }
+        },
+        'toggle-all-cells-output-scrolling':{
+            help: "Toggle the scrolling state of all output areas",
+            handler : function (env, event) {
+                env.notebook.toggle_all_output_scroll();
+            }
+        },
+
+        'clear-all-cells-output':{
```
Comment (ellisonbg) [inline notebook/static/notebook/js/actions.js]: This one is perfect.

File: notebook/static/notebook/js/actions.js [hunk 2]
```diff
@@ -563,6 +563,25 @@ define(function(require){
                 env.notebook.save_widget.rename_notebook({notebook: env.notebook});
             }
         },
+        'toggle-all-cells-output-visibility':{
```
Comment (ellisonbg) [inline notebook/static/notebook/js/actions.js]: Thinking out loud...

In the new models we have built, this is controlled by the `IOutputAreaViewModel.collapsed` property. The input area also has a collapsed property. The scrolling behavior is set by the `scrolled` property. I think the action names should follow the names of the new model properties, so:

`toggle-all-cells-output-collapsed`
`toggle-all-cells-output-scrolled`

And eventually:

`toggle-all-cells-input-collapsed`
`toggle-all-cells-input-scrolled`

If we want to change the `collapsed` property of our new models to `visible` we could change the actions here too. But I think that collapsed in a better word as there will still be some UI visible to indicate it is collapsed.

Comment (Carreau) [inline notebook/static/notebook/js/actions.js]: You are aware that you contradict yourself from already existing action you have named ?

File: notebook/static/notebook/js/menubar.js [hunk 3]
```diff
@@ -229,6 +229,9 @@ define([
             '#toggle_current_output': 'toggle-cell-output-visibility',
             '#toggle_current_output_scroll': 'toggle-cell-output-scrolling',
             '#clear_current_output': 'clear-cell-output',
```
Comment (Carreau) [inline notebook/static/notebook/js/menubar.js]: Like here is already `visibility`, and `scrolling`.

**Comment by Carreau** [comment]
Despite that you contradict yourself from before (that wouldn't be the first time :-P )

I implemented your change and did the `visibility` -> `collapsed`, `scrolling`->`scrolled` changes on other actions and places.

**Comment by ellisonbg** [comment]
LOL, yes I definitely contradict myself at times - or change my mind. Can
you clarify my inconsistency in this case? I don't intend to be
inconsistent here :)

On Thu, Nov 5, 2015 at 12:56 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> Despite that you contradict yourself from before (that wouldn't be the
> first time :-P )
>
> I implemented your change and did the visibility -> collapsed, scrolling->
> scrolled changes on other actions and places.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/692#issuecomment-154137812>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by Carreau** [comment]
well you fought in a previous issue to name the action for a single cell "visibility" and now for multiple cell you want "collapsed".

**Comment by ellisonbg** [comment]
Ahh, sorry I wasn't clear. When I did the initial action re-naming, I knew
that these actions in particular were going to need some additional work
that would probably be informed by the new model design we were doing.

@jasongrout and I had a conversation yesterday discussing if "collapsed" or
"visible" is the right word to use here and in the new models. Because the
UI will need to display *some* visible representation indicating that an
output or input is present by not fully seen, I don't think the word
"visible" quite captures it - it is visible, just not in its full
realization - in other words, it is "collapsed". So I think the actions and
models should use this name rather than visible.

Does that make sense?

On Fri, Nov 6, 2015 at 10:44 AM, Matthias Bussonnier <
notifications@github.com> wrote:

> well you fought in a previous issue to name the action for a single cell
> "visibility" and now for multiple cell you want "collapsed".
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/692#issuecomment-154442047>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by Carreau** [comment]
I was not discussing your choice, I was just pointed the inconsistency, The current state of the PR use collabsed everywhere.

**Comment by ellisonbg** [comment]
OK sounds good :)

On Fri, Nov 6, 2015 at 4:47 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> I was not discussing your choice, I was just pointed the inconsistency,
> The current state of the PR use collabsed everywhere.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/692#issuecomment-154551593>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Pull request merged**